frontend/components/hourInput.tsx

```diff
@@ -1,6 +1,62 @@
+import { ChangeEvent, FocusEvent, useState } from "react";
+
+export interface IWorkDuration {
+  hours: number;
+  minutes: number;
+}
+
+export const getWorkDuration = (s: string): IWorkDuration => {
```

Comment:
```suggestion
export const getWorkDuration = (timeString: string): IWorkDuration => {
```
I think we should avoid single-letter parameters/variables (https://www.codereadability.com/i-n-k-single-letter-variable-names/).
I would prefer `timeString`. Just `time` would suggest, that it would be a `DateTime` object and not a string. Therefore the suffix.